processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingOptions.java

```diff
@@ -480,12 +480,11 @@ public MappingControl getMappingControl(ElementUtils elementUtils) {
 
     /**
      *  mapping can only be inversed if the source was not a constant nor an expression nor a nested property
-     *  and the mapping is not a 'target-source-ignore' mapping
```

Comment:
I'm still curious why this was explicitly stated here.

Reply:
This looks like an ancient thing. The `MappingOptions` was called `Mapping` at some point. The rename was done in https://github.com/mapstruct/mapstruct/commit/7bee12138da5f601364987d053e4fd3eeb43cbcd.

The line `isIgnored && sourceName == null` was added first in https://github.com/mapstruct/mapstruct/commit/4d8bc29347e37bcf22d121829802e7f58603afbb. And looking at that commit it seems like the logic was not ported properly, because prior to that we had

```java
        // mapping can only be reversed if the source was not a constant nor an expression nor a nested property
        if ( constant != null || javaExpression != null ) {
            return null;
        }

        // should only ignore a property when 1) there is a sourceName defined or 2) there's a name match
        if ( isIgnored ) {
            if ( sourceName == null && !hasPropertyInReverseMethod( targetName, method ) ) {
                return null;
            }
        }
```

and after that we had

```java
        // mapping can only be reversed if the source was not a constant nor an expression nor a nested property
        // and the mapping is not a 'target-source-ignore' mapping
        if ( constant != null || javaExpression != null || ( isIgnored && sourceName == null ) ) {
            return null;
        }
```

I cannot say why this was the case back then

Reply:
I was curious too @zyberzebra. This looks like an ancient thing. The `MappingOptions` was called `Mapping` at some point. The rename was done in https://github.com/mapstruct/mapstruct/commit/7bee12138da5f601364987d053e4fd3eeb43cbcd.

The line `isIgnored && sourceName == null` was added first in https://github.com/mapstruct/mapstruct/commit/4d8bc29347e37bcf22d121829802e7f58603afbb. And looking at that commit it seems like the logic was not ported properly, because prior to that we had

```java
        // mapping can only be reversed if the source was not a constant nor an expression nor a nested property
        if ( constant != null || javaExpression != null ) {
            return null;
        }

        // should only ignore a property when 1) there is a sourceName defined or 2) there's a name match
        if ( isIgnored ) {
            if ( sourceName == null && !hasPropertyInReverseMethod( targetName, method ) ) {
                return null;
            }
        }
```

and after that we had

```java
        // mapping can only be reversed if the source was not a constant nor an expression nor a nested property
        // and the mapping is not a 'target-source-ignore' mapping
        if ( constant != null || javaExpression != null || ( isIgnored && sourceName == null ) ) {
            return null;
        }
```

Things were quite different back then. We are now handling the target later and differently to back then, so the fix done by @Hypnagokali looks like spot on.